includes/wikia/traits/TitleTrait.php

```diff
@@ -0,0 +1,160 @@
+	abstract function getDBkey();
+
+	/**
+	 * Returns a list of pages that link to that Title
+	 *
+	 * @param int $level How deep should you go?
+	 * @return array
+	 */
+	public function getIndirectLinks( $level = 0 ) {
+		global $wgContentNamespaces;
+
+		$rows = [];
+
+		$dbr = wfGetDB( DB_SLAVE );
+		$options = [];
+
+		$hidelinks = false;
+		$hideredirs = false;
+		$hidetrans = false;
+		$hideimages = false; //$target->getNamespace() != NS_FILE;
+
+		$fetchlinks = ( !$hidelinks || !$hideredirs );
+
+		// Make the query
+		$plConds = [
+			'page_id=pl_from',
+			'pl_namespace' => $this->getNamespace(),
+			'pl_title' => $this->getDBkey(),
+		];
+
+		if ( $hideredirs ) {
+			$plConds['rd_from'] = null;
+		} elseif ( $hidelinks ) {
+			$plConds[] = 'rd_from is NOT NULL';
+		}
+
+		$tlConds = [
+			'page_id=tl_from',
+			'tl_namespace' => $this->getNamespace(),
+			'tl_title' => $this->getDBkey(),
+		];
+
+		$ilConds = [
+			'page_id=il_from',
+			'il_to' => $this->getDBkey(),
+		];
+
+		if ( is_array( $wgContentNamespaces ) && !empty( $wgContentNamespaces ) ) {
+			$namespaces = implode( ',', $wgContentNamespaces );
+
+			$plConds[] = 'page_namespace IN (' . $namespaces . ')';
+			$tlConds[] = 'page_namespace IN (' . $namespaces . ')';
+			$ilConds[] = 'page_namespace IN (' . $namespaces . ')';
+		} elseif ( is_int( $wgContentNamespaces ) ) {
+			$plConds['page_namespace'] = $wgContentNamespaces;
+			$tlConds['page_namespace'] = $wgContentNamespaces;
+			$ilConds['page_namespace'] = $wgContentNamespaces;
+		}
+
+		// Enforce join order, sometimes namespace selector may
+		// trigger filesorts which are far less efficient than scanning many entries
+		$options[] = 'STRAIGHT_JOIN';
+
+		//$options['LIMIT'] = $queryLimit;
+		$fields = [ 'page_id', 'page_namespace', 'page_title', 'rd_from' ];
+
+		$joinConds = [
+			'redirect' => [
+				'LEFT JOIN',
+				[
+					'rd_from = page_id',
+					'rd_namespace' => $this->getNamespace(),
+					'rd_title' => $this->getDBkey(),
+					'(rd_interwiki is NULL) or (rd_interwiki = \'\')',
+				]
+			]
+		];
+
+		if ( $fetchlinks ) {
+			$options['ORDER BY'] = 'pl_from';
+			$plRes = $dbr->select(
+				[ 'pagelinks', 'page', 'redirect' ],
+				$fields,
+				$plConds,
+				__METHOD__,
+				$options,
+				$joinConds
+			);
+		}
+
+		if ( !$hidetrans ) {
+			$options['ORDER BY'] = 'tl_from';
+			$tlRes = $dbr->select(
+				[ 'templatelinks', 'page', 'redirect' ],
+				$fields,
+				$tlConds,
+				__METHOD__,
+				$options,
+				$joinConds
+			);
+		}
+
+		if ( !$hideimages ) {
+			$options['ORDER BY'] = 'il_from';
+			$ilRes = $dbr->select(
+				[ 'imagelinks', 'page', 'redirect' ],
+				$fields,
+				$ilConds,
+				__METHOD__,
+				$options,
+				$joinConds
+			);
+		}
+
+		// Read the rows into an array and remove duplicates
+		// templatelinks comes second so that the templatelinks row overwrites the
+		// pagelinks row, so we get (inclusion) rather than nothing
+		if ( $fetchlinks ) {
```

Comment:
I guess that the author wanted to break the code into logical blocks for readability... I'm with you on that matter. :)